packages/dapp/src/components/listing/ListingPhaseActions.tsx

```diff
@@ -167,41 +212,41 @@ class ListingPhaseActions extends React.Component<ListingPhaseActionsProps> {
   private renderApplicationPhase(): JSX.Element | null {
     const endTime = this.props.listing!.data.appExpiry.toNumber();
     const phaseLength = this.props.parameters.applyStageLen;
-    const approveForChallengeProgressModal = this.renderApproveForChallengeProgressModal();
-    const submitChallengeProgressModal = this.renderSubmitChallengeProgressModal();
-    const modalContentComponents = {
-      [ModalContentEventNames.IN_PROGRESS_APPROVE_FOR_CHALLENGE]: approveForChallengeProgressModal,
-      [ModalContentEventNames.IN_PROGRESS_SUBMIT_CHALLENGE]: submitChallengeProgressModal,
-    };
-    const transactions = [
-      {
-        transaction: approveForChallenge,
-        progressEventName: ModalContentEventNames.IN_PROGRESS_APPROVE_FOR_CHALLENGE,
-      },
-      {
-        transaction: this.challenge,
-        progressEventName: ModalContentEventNames.IN_PROGRESS_SUBMIT_CHALLENGE,
-      },
-    ];
-
     if (!endTime || !phaseLength) {
       return null;
     }
 
     return (
-      <InApplicationCard
-        endTime={endTime}
-        phaseLength={phaseLength}
-        transactions={transactions}
-        modalContentComponents={modalContentComponents}
-      />
+      <>
+        <InApplicationCard
+          endTime={endTime}
+          phaseLength={phaseLength}
+          handleSubmitChallenge={this.handleSubmitChallenge}
+        />
+        {this.renderSubmitChallengeModal()}
+      </>
     );
   }
 
+  private handleSubmitChallenge = () => {
+    this.setState({ isChallengeModalOpen: true });
+  };
+
+  private updateChallengeStatement = (value: any) => {
+    // Using functions here ensures that these state changes
+    // happen synchronously
+    this.setState(() => ({ isPostStatementDisabled: true }));
```

Comment:
what is `isPostStatementDisabled` used for? this function is kinda confusing

Reply:
Ah, that state value isn't being used anymore. I'll remove that code.

The 'Post Statement' and 'Submit Challenge' actions were initially separate buttons (per the mockup), and that value toggled the `disabled` attribute on the 'Post Statement' button.